Cabal/Distribution/PackageDescription/Check.hs

```diff
@@ -245,6 +246,12 @@ checkLibrary pkg lib =
       PackageDistInexcusable $
            "To use the 'required-signatures' field the package needs to specify "
         ++ "at least 'cabal-version: >= 1.21'."
+
+    -- check that all autogen-modules appear on other-modules or exposed-modules
+  , checkVersion [1,25] (not $ and $ map (flip elem (libModules lib)) (libModulesAutogen lib)) $
```

Comment:
You are right, and also there's no need to check the version because 'autogen-modules' would only parse and contain something when version >= 1.25. I can just remove that version check